app/views/events/_event.html.haml

```diff
@@ -7,6 +7,10 @@
           at
           = link_to event.venue.name, event.venue.website
     .event__details
+      - if defined?(event.cancelled) && event.cancelled?
```

Comment:
I'm not that familiar with ruby on rails yet, but I if understood correctly, events and courses are using this same haml-file. So yes I added the defined-check there to make sure nothing breaks if non-workshops don't have cancelled-field defined, but I'm not sure if that's necessary?

Reply:
Your coding is spot on - `defined?` makes complete sense as it would return an `undefined method` exception otherwise. I'm not used to it because I normally expect objects to respond to the same method calls. 

I would guess they end up cancelling other event types as well (course, meeting, event) and I would double check that they only wanted Workshop cancellable and nothing else. 

This is the first time `defined?` has been used in the code base and in all other cases Workshop, Course, Meeting and Event have behaved exactly the same - there's _just_ this one time they haven't.

Additional suspicion comes from it being a bug report rather than a design decision. Bug reporters think about their problem -  programmers think about the system.